js/apps/system/_admin/aardvark/APP/frontend/js/views/databaseView.js

```diff
@@ -176,34 +176,39 @@
       if (frontendConfig.forceOneShard) {
         sharding = 'single';
       }
-      var replicationFactor = $('#new-replication-factor').val();
-      var writeConcern = $('#new-write-concern').val();
-
-      var options = {
-        name: dbname,
-        "options" : {
-          "sharding" : sharding,
-          "replicationFactor" : Number(replicationFactor),
-          "writeConcern" : Number(writeConcern),
-        },
-        users: [{
-          username: userName
-        }]
-      };
-
-      this.collection.create(options, {
-        error: function (data, err) {
-          self.handleError(err, dbname);
-        },
-        success: function (data) {
-          if (window.location.hash === '#databases') {
-            self.updateDatabases();
+      
+      let replicationFactor = $('#new-replication-factor').val();
+      let writeConcern = $('#new-write-concern').val();
+
+      if(parseInt(writeConcern) <= parseInt(replicationFactor)) {
+        var options = {
+          name: dbname,
+          "options" : {
+            "sharding" : sharding,
+            "replicationFactor" : Number(replicationFactor),
+            "writeConcern" : Number(writeConcern),
+          },
+          users: [{
+            username: userName
+          }]
+        };
+
+        this.collection.create(options, {
+          error: function (data, err) {
+            self.handleError(err, dbname);
+          },
+          success: function (data) {
+            if (window.location.hash === '#databases') {
+              self.updateDatabases();
+            }
+            arangoHelper.arangoNotification('Database ' + _.escape(data.get('name')) + ' created.');
           }
-          arangoHelper.arangoNotification('Database ' + _.escape(data.get('name')) + ' created.');
-        }
-      });
+        });
 
-      arangoHelper.arangoNotification('Database creation in progress.');
+        arangoHelper.arangoNotification('Database creation in progress.');
+      } else {
+        arangoHelper.arangoError('DB', 'No database created. Write concerns must be smaller or equal compared to the replication factor.');
```

Comment:
Totally fine, but just asking: Why do we need that implementation now? Shouldn't the backend handle it anyway?

Reply:
Yes, this should definitely be handled in the backend, but as it currently isn't, it is better to at least check it in the frontend.